### PR TITLE
notify chat subsystem on identifies

### PR DESCRIPTION
**Pull request by heronhaye**

### CODE REVIEW

File: go/engine/identify2_with_uid.go [hunk 1]
```diff
@@ -347,6 +347,11 @@ func (e *Identify2WithUID) run(m libkb.MetaContext) {
 	err := e.runReturnError(m)
 	e.unblock(m /* isFinal */, true, err)
 
+	if !e.arg.IdentifyBehavior.IsChatIdentify() {
```
Comment (heronhaye) [inline go/engine/identify2_with_uid.go]: Need to be careful not to cause an infinite loop here.

Alternate solution is to export the results of this identify to TLFBreaks and call ` globals.CtxIdentifyNotifier(ctx)` directly from a new notification handler, without running an identify. But I thought that might end up out of sync with the chat's regular identifies which runs Identify2 with different parameters than the original call may be using.

Comment (heronhaye) [reply go/engine/identify2_with_uid.go]: Maybe safer would be to only do this on the GUI profile behavior.

Comment (mmaxim) [reply go/engine/identify2_with_uid.go]: We can always just not use `UserChanged` here

File: go/engine/identify2_with_uid.go [hunk 2]
```diff
@@ -343,10 +343,53 @@ func (e *Identify2WithUID) Run(m libkb.MetaContext) (err error) {
...
+	if libkb.IsIdentifyProofError(inErr) {
+		update.Breaks = keybase1.TLFBreak{Breaks: []keybase1.TLFIdentifyFailure{
+			keybase1.TLFIdentifyFailure{
+				User:   *e.them.Export(),
+				Breaks: res.TrackBreaks,
+			},
+		}}
+	}
+	m.Debug("Identify2WithUID.run: running HandleChatIdentifyUpdate: %#v", update)
+	m.G().NotifyRouter.HandleChatIdentifyUpdate(m.Ctx(), update)
```
Comment (heronhaye) [inline go/engine/identify2_with_uid.go]: Note this doesn't update the chat's CachingIdentifyNotifier's cache. Is that ok? If not, my thoughts are
to make another one of these listeners like HandleUserChanged, but for HandleIdentityFinished. That can update the chat's cache.

File: go/protocol/keybase1/extras.go [hunk 3]
```diff
@@ -1401,6 +1401,18 @@ func (b TLFIdentifyBehavior) SkipExternalChecks() bool {
 	}
 }
 
+// ShouldRefreshChatView indicates that when the identify is complete, we
+// should update the chat system's view of the computed track breaks (also
+// affects username coloring in the GUI).
+func (b TLFIdentifyBehavior) ShouldRefreshChatView() bool {
+	switch b {
+	case TLFIdentifyBehavior_GUI_PROFILE, TLFIdentifyBehavior_CLI:
```
Comment (heronhaye) [inline go/protocol/keybase1/extras.go]: update on both going to gui profile and IDing in CLI

**Comment by heronhaye** [comment]
made some changes, ptal @mmaxim

File: go/engine/identify2_with_uid.go (hunk 2 above)
Comment (mmaxim) [reply go/engine/identify2_with_uid.go]: Should be fine

**Comment by heronhaye** [comment]
Merging this over CI flakes.

**Pull request merged**